saving to /workspace/develop/CalibTIP/results/mobilenet_v2.adaquant
creating model mobilenet_v2
Creating absorb_bn state dict
Creating absorb_bn state dict /workspace/develop/CalibTIP/results/mobilenet_v2.adaquant/mobilenet_v2.absorb_bn
saving to /tmp/2022-05-13_18-05-22
creating model mobilenet_v2
created model with configuration: {'dataset': 'imagenet', 'batch_norm': False, 'measure': True, 'perC': True}
number of parameters: 3577420
loaded checkpoint '/workspace/develop/CalibTIP/results/mobilenet_v2.adaquant/mobilenet_v2.absorb_bn'
Traceback (most recent call last):
  File "main.py", line 1014, in <module>
    main()
  File "main.py", line 355, in main
    main_worker(args)
  File "main.py", line 635, in main_worker
    model = search_replace_layer_from_json(model, onnx_model, precision_config)
UnboundLocalError: local variable 'onnx_model' referenced before assignment
saving to /tmp/2022-05-13_18-05-27
creating model mobilenet_v2
created model with configuration: {'dataset': 'imagenet', 'batch_norm': False, 'measure': False, 'perC': True}
number of parameters: 3577420
usage: main.py [-h] [--results-dir RESULTS_DIR] [--save SAVE]
               [--onnxinput ONNXINPUT] [--datasets-dir DATASETS_DIR]
               [--dataset DATASET] [--model MODEL] [--input-size INPUT_SIZE]
               [--model-config MODEL_CONFIG] [--dtype DTYPE] [--device DEVICE]
               [--device-ids DEVICE_IDS [DEVICE_IDS ...]]
               [--world-size WORLD_SIZE] [--local_rank LOCAL_RANK]
               [--dist-init DIST_INIT] [--dist-backend DIST_BACKEND] [-j N]
               [--epochs N] [--start-epoch N] [-b N]
               [--eval-batch-size EVAL_BATCH_SIZE] [--optimizer OPT]
               [--label-smoothing LABEL_SMOOTHING] [--mixup MIXUP]
               [--duplicates DUPLICATES] [--chunk-batch CHUNK_BATCH]
               [--cutout] [--autoaugment] [--grad-clip GRAD_CLIP]
               [--loss-stepsize LOSS_STEPSIZE] [--lr LR] [--momentum M]
               [--weight-decay W] [--print-freq N]
               [--adapt-grad-norm ADAPT_GRAD_NORM] [--resume PATH] [-e FILE]
               [--seed SEED] [-ab ABSORB_BN] [-lfv LOAD_FROM_VISION]
               [--measure] [--prune] [--fine-tune]
               [--num-sp-layers NUM_SP_LAYERS]
               [--names-sp-layers [NAMES_SP_LAYERS [NAMES_SP_LAYERS ...]]]
               [--layers_precision_dict LAYERS_PRECISION_DICT]
               [--layers_precision_json LAYERS_PRECISION_JSON]
               [--keep_first_last] [--pretrained] [--extract-bias-mean]
               [--lapq] [--update_only_th] [--rec] [--kld_loss]
               [--evaluate_init_configuration] [--nbits_weight NBITS_WEIGHT]
               [--nbits_act NBITS_ACT] [--optimize_rounding]
               [--ignore-downsample] [--optimize-weights] [-i8m FILE]
               [-i4m FILE] [-i2m FILE] [--mixed_builder] [--suffix SUFFIX]
               [--adaquant] [--seq_adaquant] [--per-layer] [--mixed-builder]
               [--batch-norn-tuning] [--bias-tuning] [--eval_on_train]
               [--opt_model_paths OPT_MODEL_PATHS] [--precisions PRECISIONS]
               [--tuning-iter TUNING_ITER] [--res_log RES_LOG] [--cmp CMP]
               [--layers_precision_json_4_IP LAYERS_PRECISION_JSON_4_IP]
main.py: error: invalid checkpoint: /workspace/develop/CalibTIP/results/mobilenet_v2.adaquant/mobilenet_v2.absorb_bn.w4a8.measure_perC
saving to /workspace/develop/CalibTIP/results/mobilenet_v2.adaquant
creating model mobilenet_v2
Creating absorb_bn state dict
Creating absorb_bn state dict /workspace/develop/CalibTIP/results/mobilenet_v2.adaquant/mobilenet_v2.absorb_bn
saving to /tmp/2022-05-13_18-05-39
creating model mobilenet_v2
created model with configuration: {'dataset': 'imagenet', 'batch_norm': False, 'measure': True, 'perC': True}
number of parameters: 3577420
loaded checkpoint '/workspace/develop/CalibTIP/results/mobilenet_v2.adaquant/mobilenet_v2.absorb_bn'
Traceback (most recent call last):
  File "main.py", line 1014, in <module>
    main()
  File "main.py", line 355, in main
    main_worker(args)
  File "main.py", line 635, in main_worker
    model = search_replace_layer_from_json(model, onnx_model, precision_config)
UnboundLocalError: local variable 'onnx_model' referenced before assignment
saving to /tmp/2022-05-13_18-05-44
creating model mobilenet_v2
created model with configuration: {'dataset': 'imagenet', 'batch_norm': False, 'measure': False, 'perC': True}
number of parameters: 3577420
usage: main.py [-h] [--results-dir RESULTS_DIR] [--save SAVE]
               [--onnxinput ONNXINPUT] [--datasets-dir DATASETS_DIR]
               [--dataset DATASET] [--model MODEL] [--input-size INPUT_SIZE]
               [--model-config MODEL_CONFIG] [--dtype DTYPE] [--device DEVICE]
               [--device-ids DEVICE_IDS [DEVICE_IDS ...]]
               [--world-size WORLD_SIZE] [--local_rank LOCAL_RANK]
               [--dist-init DIST_INIT] [--dist-backend DIST_BACKEND] [-j N]
               [--epochs N] [--start-epoch N] [-b N]
               [--eval-batch-size EVAL_BATCH_SIZE] [--optimizer OPT]
               [--label-smoothing LABEL_SMOOTHING] [--mixup MIXUP]
               [--duplicates DUPLICATES] [--chunk-batch CHUNK_BATCH]
               [--cutout] [--autoaugment] [--grad-clip GRAD_CLIP]
               [--loss-stepsize LOSS_STEPSIZE] [--lr LR] [--momentum M]
               [--weight-decay W] [--print-freq N]
               [--adapt-grad-norm ADAPT_GRAD_NORM] [--resume PATH] [-e FILE]
               [--seed SEED] [-ab ABSORB_BN] [-lfv LOAD_FROM_VISION]
               [--measure] [--prune] [--fine-tune]
               [--num-sp-layers NUM_SP_LAYERS]
               [--names-sp-layers [NAMES_SP_LAYERS [NAMES_SP_LAYERS ...]]]
               [--layers_precision_dict LAYERS_PRECISION_DICT]
               [--layers_precision_json LAYERS_PRECISION_JSON]
               [--keep_first_last] [--pretrained] [--extract-bias-mean]
               [--lapq] [--update_only_th] [--rec] [--kld_loss]
               [--evaluate_init_configuration] [--nbits_weight NBITS_WEIGHT]
               [--nbits_act NBITS_ACT] [--optimize_rounding]
               [--ignore-downsample] [--optimize-weights] [-i8m FILE]
               [-i4m FILE] [-i2m FILE] [--mixed_builder] [--suffix SUFFIX]
               [--adaquant] [--seq_adaquant] [--per-layer] [--mixed-builder]
               [--batch-norn-tuning] [--bias-tuning] [--eval_on_train]
               [--opt_model_paths OPT_MODEL_PATHS] [--precisions PRECISIONS]
               [--tuning-iter TUNING_ITER] [--res_log RES_LOG] [--cmp CMP]
               [--layers_precision_json_4_IP LAYERS_PRECISION_JSON_4_IP]
main.py: error: invalid checkpoint: /workspace/develop/CalibTIP/results/mobilenet_v2.adaquant/mobilenet_v2.absorb_bn.w8a8.measure_perC
saving to /tmp/2022-05-13_18-05-47
creating model mobilenet_v2
created model with configuration: {'dataset': 'imagenet', 'batch_norm': False, 'measure': False, 'perC': True, 'depth': 50}
number of parameters: 3577420
usage: main.py [-h] [--results-dir RESULTS_DIR] [--save SAVE]
               [--onnxinput ONNXINPUT] [--datasets-dir DATASETS_DIR]
               [--dataset DATASET] [--model MODEL] [--input-size INPUT_SIZE]
               [--model-config MODEL_CONFIG] [--dtype DTYPE] [--device DEVICE]
               [--device-ids DEVICE_IDS [DEVICE_IDS ...]]
               [--world-size WORLD_SIZE] [--local_rank LOCAL_RANK]
               [--dist-init DIST_INIT] [--dist-backend DIST_BACKEND] [-j N]
               [--epochs N] [--start-epoch N] [-b N]
               [--eval-batch-size EVAL_BATCH_SIZE] [--optimizer OPT]
               [--label-smoothing LABEL_SMOOTHING] [--mixup MIXUP]
               [--duplicates DUPLICATES] [--chunk-batch CHUNK_BATCH]
               [--cutout] [--autoaugment] [--grad-clip GRAD_CLIP]
               [--loss-stepsize LOSS_STEPSIZE] [--lr LR] [--momentum M]
               [--weight-decay W] [--print-freq N]
               [--adapt-grad-norm ADAPT_GRAD_NORM] [--resume PATH] [-e FILE]
               [--seed SEED] [-ab ABSORB_BN] [-lfv LOAD_FROM_VISION]
               [--measure] [--prune] [--fine-tune]
               [--num-sp-layers NUM_SP_LAYERS]
               [--names-sp-layers [NAMES_SP_LAYERS [NAMES_SP_LAYERS ...]]]
               [--layers_precision_dict LAYERS_PRECISION_DICT]
               [--layers_precision_json LAYERS_PRECISION_JSON]
               [--keep_first_last] [--pretrained] [--extract-bias-mean]
               [--lapq] [--update_only_th] [--rec] [--kld_loss]
               [--evaluate_init_configuration] [--nbits_weight NBITS_WEIGHT]
               [--nbits_act NBITS_ACT] [--optimize_rounding]
               [--ignore-downsample] [--optimize-weights] [-i8m FILE]
               [-i4m FILE] [-i2m FILE] [--mixed_builder] [--suffix SUFFIX]
               [--adaquant] [--seq_adaquant] [--per-layer] [--mixed-builder]
               [--batch-norn-tuning] [--bias-tuning] [--eval_on_train]
               [--opt_model_paths OPT_MODEL_PATHS] [--precisions PRECISIONS]
               [--tuning-iter TUNING_ITER] [--res_log RES_LOG] [--cmp CMP]
               [--layers_precision_json_4_IP LAYERS_PRECISION_JSON_4_IP]
main.py: error: invalid checkpoint: results/mobilenet_v2.adaquant/mobilenet_v2.absorb_bn.w8a8.measure_perC.adaquant
Traceback (most recent call last):
  File "mpip_compression_pytorch_multi.py", line 369, in <module>
    sol, expect_acc, comp, policy = mpip_compression2(files=files, replace_precisions=replace_precisions, Degradation=Deg, noise=sigma, method=ip_method, bops=args.ip_gain)
  File "mpip_compression_pytorch_multi.py", line 140, in mpip_compression2
    data[prec] = pd.read_csv(f)
  File "/opt/conda/lib/python3.8/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 680, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 575, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 933, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1217, in _make_engine
    self.handles = get_handle(  # type: ignore[call-overload]
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/common.py", line 789, in get_handle
    handle = open(
FileNotFoundError: [Errno 2] No such file or directory: 'results/mobilenet_v2.adaquant/mobilenet_v2.absorb_bn.w8a8.measure_perC.adaquant.per_layer_accuracy.A8.W4.csv'
